src/main/java/frc/robot/commands/CmdArmReset.java

```diff
@@ -0,0 +1,34 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Arm;
+
+public class CmdArmReset extends CommandBase 
+{
+    private final Arm _armSubsystem;
+
+    public CmdArmReset() 
+    {
+        _armSubsystem = Arm.getInstance();
+    }
+
+    @Override
+    public void execute() 
+    {
+        _armSubsystem.setExtensionMotorSpeed(-0.5);
+    }
+
+    @Override
+    public boolean isFinished() 
+    {
+        if (_armSubsystem.isLimitSwitchPressed())
+        {
+            _armSubsystem.setExtensionMotorSpeed(0);
+            return true;
+        }
+        else
+        {
+            return false;
+        }
```

Comment:
In the `isFinished` function, we don't want to perform any control logic (for example, turning motors on/off). This function should just be used to determine whether the command should end.

You have this with the `if (_armSubsystem.isLimitSwitchPressed())` check. Since we don't want line 26, we can shorten this whole function up to simply be:

```java
@Override
public boolean isFinished()
{
    return _armSubsystem.isLimitSwitchPressed();
}
```

We do, however, want to shut off the extension motor when the command is over. This is what you're trying to do on line 26, but this should happen in a new function called `public void end(boolean interrupted)`. This function will automatically be called once when the command finishes. In the `end` function, you'll also want to reset the extension encoder position to 0. You'll need to create a new function to do this in the Arm subsystem.